man/preupg.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.6.
.TH PREUPGRADE "5" "September 2018" "Preupgrade Assistant 2.6.1" "User Commands"
.SH NAME
Preupgrade Assistant - advises on the feasibility of system upgrade or migration
.SH SYNOPSIS
preupg [[-h|--help] | [--version] | [--cleanup] | [-l|--list-contents-set]]

preupg [-s|--scan MODULE_SET] | [-c|--contents ALL_XCCDF_PATH] [-d|--debug] [-S|--skip-common] [-m|--mode MODE] [--force] [--text] [--dst-arch ARCH] [--old-report-style] [--select-rules RULES] [-v|--verbose]

preupg --list-rules [[-s|--scan MODULE_SET] | [-c|--contents ALL_XCCDF_PATH]]

preupg --riskcheck [-v|--verbose]

preupg -u|--upload URL [-r|--results TARBALL]
.SH DESCRIPTION
The Preupgrade Assistant is a framework designed to run the Preupgrade Assistant modules, which analyze the system for possible in-place upgrade limitations. It is based on a modular system, with each module performing a separate test, checking for package removals, incompatible obsolete packages, changes in libraries, users, groups, services, or incompatibilities of command-line options or configuration files. It is able to execute post-upgrade scripts to finalize complex tasks after the system upgrade. Apart from performing the in-place upgrades, the Preupgrade Assistant is also capable of migrating the system. It then produces a report, which assists you in performing the upgrade itself by outlining potential problem areas and by offering suggestions about mitigating any possible incompatibilities. The Preupgrade Assistant utility is a Red Hat Upgrade Tool prerequisite for completing a successful in-place upgrade to the next major version of Red Hat Enterprise Linux.
.SH OPTIONS
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
Show help message and exit.
.TP
\fB\-S\fR, \fB\-\-skip\-common\fR
Skip generating files containing information about the
system. For assessing the system these files are
needed (by modules) but in the case the system remains
the same (the same installed packages, configuration
files not touched, etc.) they can be reused from the
previous runs of Preupgrade Assistant.
.TP
\fB\-d\fR, \fB\-\-debug\fR
Turn on debugging mode.
.TP
\fB\-u\fR, \fB\-\-upload\fR
Upload a system assessment result to Preupgrade
Assistant WEB\-UI.
.TP
\fB\-r\fR TARBALL, \fB\-\-results\fR=\fI\,TARBALL\/\fR
Provide path to a system assessment result tarball
which is to be uploaded to WEB\-UI. By default, the
result tarballs can be found in \fI\,/root/preupgrade\/\fP.
.TP
\fB\-l\fR, \fB\-\-list\-contents\-set\fR
List all the available sets of modules. They are
searched for in \fI\,/usr/share/preupgrade\/\fP.
.TP
\fB\-s\fR MODULE_SET, \fB\-\-scan\fR=\fI\,MODULE_SET\/\fR
Provide name of the set of modules which are to be
used for assessing the system. By default, if there is
just one set in \fI\,/usr/share/preupgrade\/\fP, Preupgrade
Assistant uses that one. Use \fB\-\-list\-contents\-set\fR
option to get a list of possible values.
.TP
\fB\-c\fR ALL_XCCDF_PATH, \fB\-\-contents\fR=\fI\,ALL_XCCDF_PATH\/\fR
Provide path to all\-xccdf.xml of the set of modules
which is to be used for assesing the system. By
default, if there is just one set in
\fI\,/usr/share/preupgrade\/\fP, Preupgrade Assistant uses that
one. Option \fB\-\-scan\fR works similarly.
.TP
\fB\-\-riskcheck\fR
Return the highest reported level of risk or result
related to system upgrade. Run Preupgrade Assistant
first \- assessment of the system needs to be performed
before using this option. When this option is used in
concert with \fB\-\-verbose\fR option, summary of the risks
are printed to STDOUT. If the \fB\-\-verbose\fR option is used
once, just HIGH and EXTREME risks are printed. If it
is used twice, all the risks are printed.
Return codes:
0 ... SLIGHT or MEDIUM risk or needs_inspection,
fixed, informational, not_applicable, not_selected,
not_checked or pass result.
1 ... HIGH risk or needs_action result.
2 ... EXTREME risk or error or fail result.
.TP
\fB\-\-force\fR
Suppress user interaction.
.TP
\fB\-\-text\fR
Generate plain text assessment report alongside XML
and HTML reports. The text report is converted from
HTML using elinks, lynx or w3m tool.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Show more information during the assessment.
.TP
\fB\-\-cleanup\fR
Remove all the files created by previous runs of
Preupgrade Assistant.
.TP
\fB\-m\fR MODE, \fB\-\-mode\fR=\fI\,MODE\/\fR
Select what you plan to do with the system after
performing its assessment by Preupgrate Assistant \-
migration or upgrade. Both modes are selected by
default. This option may only affect behaviour of the
modules \- they can provide different results when only
one mode is selected. Use one of these values:
migrate, upgrade. It may be that modules behave the
same no matter what mode is selected.
.TP
\fB\-\-select\-rules\fR=\fI\,RULES\/\fR
Execute just a subset of modules out of a module set.
Multiple modules are to be separated by a comma.
.TP
\fB\-\-list\-rules\fR
List all the modules available within a module set.
.TP
\fB\-\-dst\-arch\fR=\fI\,ARCH\/\fR
Specify an architecture of the system to be migrate
to. Available option are: x86_64, ppc64. Use of the
option is expected on 32\-bit systems as by the release
of RHEL 7, 32\-bit hardware support has been dropped.
.TP
\fB\-\-old\-report\-style\fR
Generate report with simpler style than the default.
.SH "RETURN CODES"
Possible return codes are:

0 - preupg works properly and modules exit with the following results: PASS, FIXED, INFORMATIONAL, NEEDS_INSPECTION, NOT_APPLICABLE.

1 - preupg works properly but at least one module finishes with NEEDS_ACTION result.

2 - preupg works properly but at least one module finishes with FAIL or ERROR result.

20 - preupg cannot find a proper upgrade scenario.

21 - preupg is executed with options --mode and --select-rules. This is not allowed.

22 - preupg is executed with an invalid command line option.

23 - preupg has to be executed as root.

24 - System assessment has to be performed first before using --riskcheck option.

25 - openscap and openscap-engine-sce are not installed on the system.

26 - You specified a text convertor which is not allowed. Allowed convertors are w3m, lynx and elinks.

27 - The /usr/share/preupgrade/data/preassessment/scripts.txt file is missing. The Preupgrade Assistant is not installed properly and needs to be reinstalled.

28 - We have detected some troubles with sending the report to WEB-UI.

29 - Internal problem.

30 - User aborted the assesment.
.SH EXAMPLES
preupg -u http://example.com:8099/submit/ -r /root/preupgrade-results/preupg_results-161108001514.tar.gz

preupg -s RHEL6_7

preupg -c /usr/share/preupgrade/RHEL6_7/all-xccdf.xml

preupg --select-rules xccdf_preupg_rule_networking_vsftpd_check,xccdf_preupg_rule_networking_bind_configuration_check
.SH "REPORTING BUGS"
Preferably use https://github.com/upgrades-migrations/preupgrade-assistant/issues OR try to contact us on a freenode.net IRC channel #preupgrade.


preupg.bash
# bash completion for preupgrade-assistant
_preupg_scan() {
    echo $(cd /usr/share/preupgrade; ls -d *)
}

_preupg_result(){
    echo $(cd /root/preupgrade-results; ls -1 *.tar.gz)
}

_preupg() {
    # read output and returncode of the command with "-h"
    local opts cur prev comps
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    opts="-s --scan -v --verbose -d --debug --skip-common -u --upload -r --results --list-contents-set -c --contents
     -a --apply --riskcheck --force --text -m --mode --cleanup --select-rules --list-rules"

    #echo "SS${COMP_CWORD} and ${COMP_WORDS} and ${prev} and ${cur}SS"
    if [[ ${COMP_CWORD} == 1 && ${COMP_WORDS} == "preupg" ]]; then
        COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
        return 0
    fi
    if [[ ${COMP_CWORD} == 1 && ${cur} == -* ]] ; then
        COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
        return 0
    fi

    comps=""
    case "${prev}" in
        "-s"|"--scan")
        comps=$(_preupg_scan)
        ;;
        "-u"|"--upload")
        comps="http://127.0.0.1:8099/submit/"
        ;;
        "-r"|"--results")
        comps=$(_preupg_result)
        ;;
        "--riskcheck")
        opts="-v --verbose -d --debug"
        comps="$opts"
        ;;
        *)
        prev="${COMP_WORDS[COMP_CWORD-2]}"
        case "${prev}" in
            "-s"|"--scan")
            opts="-v --verbose -d --debug --skip-common --riskcheck --force --text"
            comps="$opts"
            ;;
            "-u"|"--upload")
            opts="-v --verbose -d --debug -r --results"
            comps="$opts"
            ;;
            "-r"|"--results")
            opts="-v --verbose -d --debug -u --upload"
            comps="$opts"
            ;;
            *)
            result=$( compgen -A file -- "$cur" )
            compopt -o filenames
            ;;
        esac
        #echo "$opts and $comps"
        COMPREPLY=( $(compgen -W "${comps}" -- "${cur}" ) )
        ;;
    esac

    COMPREPLY=( $(compgen -W "${comps}" -- "${cur}" ) )
    return 0

}
complete -F _preupg -o filenames preupg
# Local variables:
# mode: shell-script
# sh-basic-offset: 4
# sh-indent-comment: t
# indent-tabs-mode: nil
# End:
# ex: ts=4 sw=4 et filetype=sh
